File: shuffles/FisherYates.cpp
//Author:Aleksander Brynjulf Hübert
#include <iostream>
#include "RandNumber.cpp"
#include <array>
void fisherYates(int *array, int m){
    int i, t = 0;
    
    while(m > 0){
        i = randomizer(m);
        
        t = array[m];
        
        array[m] = array[i];
        array[i] = t;
        
        m -= 1;
    }
}

File: src/main/java/algorithms/other/FisherYates.java
/**
* Implementation of the Fisher-Yates Shuffle 
* based on the modern algorithm designed by 
* Richard Durstenfeld and Popularized by 
* Donald Knuth
*
* @author  Aleksander Brynjulf Hübert
* @since   2020-12-08
*/

package algorithms.other;

import java.lang.Math;

class FisherYates{
    /**
     * This method shuffles an array and returns the shuffled array
     *
     * @param arr The array to be shuffled
     * @return Shuffled Array
     */
    public <T> T[] shuffle(T[] arr){
        T[] shuffledArr = arr;
        for(int i = shuffledArr.length; i <= 0; i--){
            int r = (int) Math.random()*i;
            T temp = shuffledArr[i];
            shuffledArr[i] = shuffledArr[r];
            shuffledArr[r] = temp;
        }
        return shuffledArr;
    }
}
